apps/ledger-live-desktop/src/renderer/families/canton/PendingTransactions/PendingTransactionDetails.tsx

```diff
@@ -0,0 +1,287 @@
+import FormattedVal from "~/renderer/components/FormattedVal";
+import IconCheck from "~/renderer/icons/Check";
+import IconCross from "~/renderer/icons/Cross";
+import { CantonAccount } from "@ledgerhq/live-common/families/canton/types";
+import {
+  OpDetailsSection,
+  OpDetailsTitle,
+  OpDetailsData,
+  B,
+  HashContainer,
+  GradientHover,
+} from "~/renderer/drawers/OperationDetails/styledComponents";
+import { SplitAddress } from "~/renderer/components/OperationsList/AddressCell";
+import CopyWithFeedback from "~/renderer/components/CopyWithFeedback";
+
+type PendingProposal = {
+  contract_id: string;
+  sender: string;
+  receiver: string;
+  amount: BigNumber;
+  memo: string;
+  expires_at_micros: number;
+  isExpired: boolean;
+};
+
+type Props = {
+  onClose?: () => void;
+  account: Account;
+  contractId: string;
+  onOpenModal: (contractId: string, action: "accept" | "reject" | "withdraw") => void;
+};
+
+const PendingTransactionDetails: React.FC<Props> = ({
+  account,
+  contractId,
+  onClose,
+  onOpenModal,
+}) => {
+  const { t } = useTranslation();
+  const unit = useAccountUnit(account);
+
+  const cantonAccount = account as CantonAccount;
+  const proposal = useMemo<PendingProposal | null>(() => {
+    const pendingTransferProposals = cantonAccount?.cantonResources?.pendingTransferProposals || [];
+    const found = pendingTransferProposals.find(p => p.contract_id === contractId);
+    if (!found) return null;
+
+    const now = Date.now();
+    const isExpired = now > found.expires_at_micros / 1000;
+    return {
+      ...found,
+      isExpired,
+      amount: new BigNumber(found.amount),
+    };
+  }, [cantonAccount, contractId]);
+
+  const handleAcceptOffer = useCallback(
+    (contractId: string) => {
+      onOpenModal(contractId, "accept");
+      // Close drawer after modal state is set
+      setTimeout(() => {
+        onClose?.();
+      }, 0);
+    },
+    [onClose, onOpenModal],
+  );
+
+  const handleRejectOffer = useCallback(
+    (contractId: string) => {
+      onOpenModal(contractId, "reject");
+      // Close drawer after modal state is set
+      setTimeout(() => {
+        onClose?.();
+      }, 0);
+    },
+    [onClose, onOpenModal],
+  );
+
+  const handleWithdrawOffer = useCallback(
+    (contractId: string) => {
+      onOpenModal(contractId, "withdraw");
+      // Close drawer after modal state is set
+      setTimeout(() => {
+        onClose?.();
+      }, 0);
+    },
+    [onClose, onOpenModal],
+  );
+
+  const formatDate = useCallback((timestamp: number) => {
+    const date = new Date(timestamp / 1000);
+    return (
+      date.toISOString().split("T")[0] + " " + date.toTimeString().split(" ")[0].substring(0, 5)
+    );
+  }, []);
+
+  const [timeRemaining, setTimeRemaining] = useState<string>("");
+
+  useEffect(() => {
+    if (!proposal || proposal.isExpired) {
+      setTimeRemaining("");
+      return;
+    }
+
+    const updateTimeRemaining = () => {
```

Comment:
the logic of this function can be external to the component, improving readability.
```
const remainingTime = getRemainingTime()
if (!remainingTime) {
   setTimeRemaining("");
}
setTimeRemaining(remainingTime);
```